### DISCUSSION TITLE
Better documentation

### DISCUSSION THREAD

**Discussion by KaruroChori** [saghul/txiki.js#682] [Ideas]
One of the main issues of the current docs, is that they only cover what is reasonable to directly attach to typescript code.  
A proper `docs` subfolder with information about the internal architecture, how to embed it etc would be great for adoption.  

I recently had to bite the bullet, and handle it on several of my projects.   
For example:
- [vs.fltk](https://github.com/KaruroChori/vs-fltk) has hybrid docs generated via doxygen and mkdocs (for any explicit markdown content).
- [comfybun](https://github.com/KaruroChori/comfyui-bun-client) has hybrid docs generated via typedoc for the ts/js things, and mkdocs for the rest of the markdown entries.

For txiki, I think doxygen, mkdocs and typedoc (which is already covered) would be needed to offer full coverage.  
As I already got it sorted out for the most part, if you are interested I can put together a PR with the basic configuration & extend the CI to cover the additional generation.

**Comment by saghul** [comment]
Hey there! Thanks for bringing this up!

I started a similar effort, but got a bit sidetracked... Let's get into it.

My first candidate was Docusaurus, since I am quite familiar with it, and there is a typedoc plugin, so we could integrate it right in. I am, however, not a huge fan of how it's rendered inside Docusaurus, specially the code blocks.

<img width="2672" alt="Screenshot 2025-01-06 at 10 34 18" src="https://github.com/user-attachments/assets/6173f2fe-4ad8-4b63-b149-f5d5c0a12e26" />

I suppose you mentioned Doxygen for the C API? I suppose we could do that yes.

So it looks like we have a combination of 3 things:

- typedoc for the TS API
- doxygen for the C API
- some documentation side to link the 2 above and have some "Getting started" and such

I'm biased towards Docusaurus because it's the one I introduced in QuickJS-NG (and unrelated, but we also use it successfully with Jitsi Meet) and seems flexible enough. We could just deploy typedoc and doxygen to subdirectories and that should do it.

Thoughts?

**Reply by KaruroChori** [reply]
Just a quick reply, so I will be brutally honest :D: I think docusaurus is one of the worst offenders out there.    
It deploys the documentation as a single page application in place of a static site generation (so unlike typedoc, doxygen and virtually everything else).  
This means that:
- One is basically forced to read it through a modern browser with JS enabled. 
- Searching in the generated docs via command line tools would be a nightmare as the granularity of files is lost.  

Sure, one would still have the source markdown files, but it is a much downgraded experience when reading them.  

The added value of docusaurus is to enable using react in the docs... which I think is a very bad idea :sweat:   

By the way, I agree that style is quite bad looking compared to typedoc. Too much whitespace, data looks very sparse.

**Reply by KaruroChori** [reply]
Aside for my personal grudge against docusaurus, the combination you described is what I was also thinking.

**Reply by saghul** [reply]
Agree to disagree on Docusaurus :-) Since no documentation other than TypeDoc exists currently, and whichevere system we choose will be markdown files. Let's give mkdocs a try, I'm open to being impressed by it!

Do you want to get the ball rolling since you're more familiar with it?

**Reply by KaruroChori** [reply]
Ok, I will prepare a demo over the weekend.

**Reply by saghul** [reply]
Thank you!

**Comment by lukasMega** [comment]
Hello!
I just get 404 in documentation: 

1. searching "platform"

<img width="1190" height="871" alt="scr 2026-06-01 o 9 49 53" src="https://github.com/user-attachments/assets/c022fd21-b73f-4bb6-97e2-1a5ef1b2b369" />


2. click on search result for Platform -> https://txikijs.org/docs/api/global.tjs.system.Variable.platform/

<img width="1298" height="831" alt="scr 2026-06-01 o 9 49 59" src="https://github.com/user-attachments/assets/d8d0d35b-1661-4f84-8e17-9950ed24d034" />

**Reply by saghul** [reply]
Thanks for the heads up, I'll take a look!

**Reply by saghul** [reply]
Found the problem, the Algolia crawler was stopped because it had somehow failed domain verification. It's reindexing now and that result should no longer show up.
